### ISSUE TITLE
Duplicate header names cause parsing to overwrite data

### ISSUE THREAD

**Issue by jannah** [mholt/PapaParse#129]
I am testing a system I am building using PapaParse. One scenario I am looking it is if more than one column shared the same header. What I am seeing is that only the last column would be included in the result.
http://jannah.github.io/MetaReader
Is there a way to rename the duplicate columns?

**Comment by Overdrivr** [comment]
That's not so rare, happened to me yesterday ;)
For data pipelines, this is very annoying because it silently fails (in my case, the column in question was an "id" one, breaking all relationships in my data model).
At the bare minimum, would it be acceptable to have an error returned because fundamentally the behavior is undefined ?

**Comment by fortydegrees** [comment]
Likewise had this issue. I crudely applied the fix in @jannah 's fiddle to the parse function in papaparse.js

Line 1460, old:

```
if (fastMode || (fastMode !== false && input.indexOf(quoteChar) === -1))
			{
				var rows = input.split(newline);
				for (var i = 0; i < rows.length; i++)
				{
					row = rows[i];
					cursor += row.length;
					if (i !== rows.length - 1)
						cursor += newline.length;
					else if (ignoreLastRow)
						return returnable();
					if (comments && row.substring(0, commentsLen) === comments)
						continue;
					if (stepIsFunction)
					{
						data = [];
						pushRow(row.split(delim));
						doStep();
						if (aborted)
							return returnable();
					}
					else
						pushRow(row.split(delim));
												
					if (preview && i >= preview)
					{
						data = data.slice(0, preview);
						return returnable(true);
					}
				}
				return returnable();
			}
```

Updated to append "_" to duplicate fields:

```
if (fastMode || (fastMode !== false && input.indexOf(quoteChar) === -1))
			{
				var rows = input.split(newline);
				for (var i = 0; i < rows.length; i++)
				{
					row = rows[i];
					cursor += row.length;
					if (i !== rows.length - 1)
						cursor += newline.length;
					else if (ignoreLastRow)
						return returnable();
					if (comments && row.substring(0, commentsLen) === comments)
						continue;
					if (stepIsFunction)
					{
						data = [];
						pushRow(row.split(delim));
						doStep();
						if (aborted)
							return returnable();
					}
					else

						if (i == 0){
							let separator = '_';
							let headers = row.split(delim) //headers array
							let header_map = []; //map of headers after resolving duplicates
							let header_count = {}; //count the headers
							//process headers
							for (var i in headers) {
								var header = headers[i]
								//check if header already exists
								if (typeof header_count[header] === 'undefined') {
								//print('new column ' + header);
								header_count[header] = 1;
								} else {
									//rename header if existed starting with _1
									var current_count = header_count[header];
									header_count[header] += 1;
									//append counter to header
									let temp_header = header + separator;
									//in case the porposed name already exists in the original csv
									while (headers.indexOf(temp_header + current_count) > -1) {
										temp_header += separator
										//print(temp_header)
									}
									header = temp_header + current_count;
									header_count[header] = 1;
								}
								//add new header to header_map
								header_map.push(header);
							}
							pushRow(header_map);
						}else{
							pushRow(row.split(delim));
						}
												
					if (preview && i >= preview)
					{
						data = data.slice(0, preview);
						return returnable(true);
					}
				}
				return returnable();
			}
```

I don't see the downside to adding an allowDuplicates config flag to enable this behaviour in the standard library.

**Comment by pokoli** [comment]
Could you upload a PR with a test case that reproduces the error and a correction for it? Thanks

**Comment by fortydegrees** [comment]
I've submitted a pull request with a test case & correction: #956 

I tried to add a renameDuplicateHeaders config option but left it out as I am doing the processing in the Parse.parse function where I don't have the ability to read the config. I'm not sure it fits the styling of the codebase to modify the input outside of that function so have left it out.

There was a bug in my original comment with the processing happening at the wrong time, but have since edited it so that the bug no longer occurs.
